controllers/components/search/search.jsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import Navbar from '../navbar.jsx';
+
+class Search extends React.Component {
+    constructor( props ) {
+        super( props );
+
+        this.state = {
+            isSearch: false,
+            result: 'no data'
+        };
+    }
+    componentWillMount() {
+        const that = this;
+        function getQueriesFromURL( query, url ) {

```

Comment:
Do not ever, never, define functions within react lifecycle event handlers, it'll be defined each time the function is run.

Reply:
When a component renders, Doesn't `componentWillMount()` runs only once, before `render` gets called?
The point is it doesn't matter whether defining functions within it or not, it will run just once.
am right, or by convention it shouldn't?

Reply:
Be convention you shouldn't, yes. And the component might render multiple times, so that's why. I'm not entirely sure if the mounting happens on each render though. Still, better to define it as a separate function.